src/test/java/io/vertx/ext/eventbus/bridge/tcp/TcpEventBusBridgeTest.java

```diff
@@ -319,6 +352,86 @@ public void testUnRegister(TestContext context) {
 
   }
 
+  @Test
+  public void testReplyFromClient(TestContext context) {
+    // Send a request from java and get a response from the client
+    NetClient client = vertx.createNetClient();
+    final Async async = context.async();
+    final String address = "test";
+    client.connect(7000, "localhost", conn -> {
+      context.assertFalse(conn.failed());
+
+      NetSocket socket = conn.result();
+
+      final FrameParser parser = new FrameParser(parse -> {
+        context.assertTrue(parse.succeeded());
+        JsonObject frame = parse.result();
+        if ("message".equals(frame.getString("type"))) {
+          context.assertEquals(true, frame.getBoolean("send"));
+          context.assertEquals("Vert.x", frame.getJsonObject("body").getString("value"));
+          FrameHelper.sendFrame("send", frame.getString("replyAddress"), new JsonObject().put("value", "You got it"), socket);
+        }
+      });
+
+      socket.handler(parser);
+
+      FrameHelper.sendFrame("register", address, null, socket);
+
+      // There is now way to know that the register actually happened, wait a bit before sending.
+      vertx.setTimer( 500L, timerId -> {
+          vertx.eventBus().send(address, new JsonObject().put("value", "Vert.x"), (AsyncResult<Message<JsonObject>> respMessage) -> {
```

Comment:
use this instead on send generic instead of cast : `eventBus().<JsonObject>send(...)`